pkg/cmd/roachprod-microbench/executor.go

```
@@ -410,13 +448,24 @@ func (e *executor) executeBenchmarks() error {
 				fmt.Println()
 			}
 			tag := fmt.Sprintf("%d", logIndex)
+			timeout := false
 			if response.ExitStatus == 124 || response.ExitStatus == 137 {
 				tag = fmt.Sprintf("%d-timeout", logIndex)
+				timeout = true
 			}
 			err = report.writeBenchmarkErrorLogs(response, tag)
 			if err != nil {
 				e.log.Errorf("Failed to write error logs - %v", err)
 			}
+
+			if e.postIssues && benchmarkResponse.key == e.postConfig.binary {
+				artifactsDir := fmt.Sprintf("%s/%s", e.outputDir, benchmarkResponse.key)
+				formatter, req := createBenchmarkPostRequest(artifactsDir, response, timeout)
+				err = postBenchmarkIssue(context.Background(), e.log, formatter, req)
+				if err != nil {
+					e.log.Errorf("Failed to post benchmark issue - %v", err)
```

This is effectively a silent error since it won't fail build [1].

[1] https://github.com/cockroachdb/cockroach/issues/147116

---

Unless we return an error from `executeBenchmarks` when `errorCount > 0`? Except the counter doesn't appear to be used.

---

Good point! We don't fail execution when benchmarks fail (anymore).
But if a post error happens I think it would be good to fail the build so there's visibility. I'll change this before merging.